File: output/copilot/python/results/q9-pascals-triangle-ii.py
# https://leetcode.com/problems/pascals-triangle-ii/

class Solution(object):
    def getRow(self, rowIndex):
        """
        :type rowIndex: int
        :rtype: List[int]
        """
        if rowIndex == 0:
            return [1]
        elif rowIndex == 1:
            return [1,1]
        else:
            prev = [1,1]
            for i in range(2,rowIndex+1):
                curr = [1]
                for j in range(1,i):
                    curr.append(prev[j-1]+prev[j])
                curr.append(1)
                prev = curr
            return curr

File: output/copilot/c/results/q9-pascals-triangle-ii.c
/** https://leetcode.com/problems/pascals-triangle-ii/ */



/**
 * Note: The returned array must be malloced, assume caller calls free().
 */
int* getRow(int rowIndex, int* returnSize){
    int* res = (int*)malloc(sizeof(int) * (rowIndex + 1));
    *returnSize = rowIndex + 1;
    res[0] = 1;
    for (int i = 1; i <= rowIndex; i++) {
        res[i] = 1;
        for (int j = i - 1; j > 0; j--) {
            res[j] += res[j - 1];
        }
    }
    return res;
}
    